hbase_pipeline/src/main/java/com/splicemachine/derby/hbase/SpliceIndexObserver.java

```diff
@@ -146,8 +146,10 @@ public void stop(CoprocessorEnvironment e) throws IOException {
             super.stop(e);
             if (region != null)
                 region.close();
-            if(service!=null)
+            if(service!=null) {
                 service.shutdown();
+                DatabaseLifecycleManager.manager().deregisterGeneralService(service);
```

Comment:
If `service.shutdown()` throws an exception, deregistration will not occur. Perhaps the line above should go into a `finally` block instead (along with a null check of `service`), if you want deregistration to occur regardless.